qsimcirq/qsim_simulator.py

```diff
@@ -718,3 +857,28 @@ def _translate_circuit(
             self._translated_circuits.append((circuit, translated_circuit))
 
         return translated_circuit
+
+    def _translate_moments(
```

Comment:
There is no circuit memoization, right?

Reply:
Memoization occurs in `_translate_circuit`, which this calls for each moment. This has the added benefit that repeated moments will use their memoized translations. (Note however that this does not affect `CircuitOperation`s with repetitions, as a given `CircuitOperation` exists inside a single moment)